scripts/analysis.py

```diff
@@ -250,229 +116,198 @@ def get_sim_time_duration(cursor):
+            value_timeseries.append(value * multiplyby)
     else:
-        resources = cur.execute(exec_string(agent_ids,
-                                            'transactions.senderId',
-                                            'sum(quantity), time, qualid')
-                                + ' GROUP BY time, qualid').fetchall()
-
-    compositions = cur.execute('SELECT qualid, nucid, massfrac\
-                               FROM compositions').fetchall()
+        for i in range(0, duration):
+            value = sum(array[array[:, 0] == i][:, 1])
+            value_timeseries.append(value * multiplyby)
 
-    init_year, init_month, duration, timestep = get_sim_time_duration(cur)
-    isotope, mass, time_list = isotope_mass_time_list(resources, compositions)
+    return value_timeseries
 
-    waste_dict = get_waste_dict(isotope, mass, time_list, duration)
 
-    if influx_bool is False:
-        stacked_bar_chart(waste_dict, timestep,
-                          'Years', 'Mass [kg]',
-                          title, outputname, init_year)
-    else:
-        multi_line_plot(waste_dict, timestep,
-                        'Years', 'Mass [kg]',
-                        title, outputname, init_year)
+"""GETTERS"""
 
 
-def commodity_from_facility(cursor, facility, commodity):
-    """ Returns timeseries of commodity outflux from facility
+def snf(cursor):
+    """returns a dictionary of isotopics in sink at the end of simulation
 
     Parameters
     ----------
     cursor: sqlite cursor
         sqlite cursor
-    facility: str
-        name of facility type
-    commodity: list
-        list of commodities
 
     Returns
     -------
-    dictionary of timeseries of mass outflux of commodity from facility
+    snf_dict: dictionary
+        dirctionary with key: isotope, and value: mass
     """
+    sink_id = get_agent_ids(cur, 'sink')
 
-    cur = cursor
-    init_year, init_month, duration, timestep = get_sim_time_duration(cur)
-    commodity_dict = collections.OrderedDict()
-    for x in commodity:
-        y = ['"' + x + '"']
-        resources = cur.execute(exec_string(y, 'commodity',
-                                            'quantity, senderid, time')).fetchall()
-
-        timeseries = get_from_facility(cur, facility, duration, resources)
-        commodity_dict[x] = timeseries
+    # get list of sum(quantity) and qualid for snf
+    snf_inventory = cursor.execute(exec_string(sink_id,
+                                               'transactions.receiverId',
+                                               'sum(quantity), qualid') +
+                                   ' GROUP BY qualid').fetchall()
+    compositions = cursor.execute('SELECT qualid, nucid, massfrac '
+                                  'FROM compositions').fetchall()
+    snf_dict = collections.defaultdict(float)
+    for comp in compositions:
+        for num in snf_inventory:
+            if num[0] == comp[0]:
```

Comment:
I agree that accessing using names would be a more convinient and readable method.
I will apply this change right away.
Thank you.